### PR TITLE
Use /msgroom in more places

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

File: server/chat-commands/core.ts [hunk 1]
```diff
@@ -485,13 +485,15 @@ export const commands: ChatCommands = {
...
-	async msgroom(target, room, user, connection) {
+	async msgroom(target, room, user, connection, cmd) {
 		const [targetId, message] = Utils.splitFirst(target, ',').map(i => i.trim());
 		if (!targetId || !message) {
 			return this.parse(`/help msgroom`);
 		}
 		const targetRoom = Rooms.search(toID(targetId));
-		if (!targetRoom) return this.errorReply(`Room not found`);
+		if (!targetRoom) return this.errorReply(`Room not found.`);
+		if (!targetRoom.users[user.id]) return this.errorReply(`You must be in the room "${targetRoom.title}" to message it.`);
```
Comment (Zarel) [inline server/chat-commands/core.ts:495]: I don't think this is necessary; I think this just makes it harder on Pages. Presumably moderation commands should still work when you're not in a room.

Comment (KrisXV) [reply server/chat-commands/core.ts:495]: fair

Comment (KrisXV) [reply server/chat-commands/core.ts]: Main thing i don't like about this is you can currently use it to just send casual messages like 'hi' to rooms you're not in

File: server/chat-commands/core.ts [hunk 2]
```diff
@@ -485,13 +485,15 @@ export const commands: ChatCommands = {
...
+		if (message.startsWith(`/${cmd}`)) return this.errorReply(`You may not use /msgroom to parse itself in other rooms.`);
```
[the remark this answers is not in the document]
Comment (KrisXV) [reply server/chat-commands/core.ts]: you're right; i tried fiddling with `this.handler.name` and `this.handler.aliases` but there's little need for this to get an alias, so i just hardcoded it

File: server/chat-commands/core.ts [hunk 3]
```diff
@@ -491,7 +491,10 @@ export const commands: ChatCommands = {
 			return this.parse(`/help msgroom`);
 		}
 		const targetRoom = Rooms.search(toID(targetId));
-		if (!targetRoom) return this.errorReply(`Room not found`);
+		if (!targetRoom) return this.errorReply(`Room not found.`);
+		if (message.trim().startsWith('/msgroom')) {
```
Comment (Zarel) [inline server/chat-commands/core.ts:495]:
```suggestion
		if (message.trim().startsWith('/msgroom ')) {
```

File: server/chat-commands/core.ts [hunk 4]
```diff
@@ -491,7 +491,10 @@ export const commands: ChatCommands = {
...
+			return this.errorReply(`You may not use /msgroom to parse itself in other rooms.`);
```
Comment (Zarel) [inline server/chat-commands/core.ts:496]:
```suggestion
			return this.errorReply(`Please do not nest /msgroom inside itself.`);
```

**Review by Zarel — APPROVED** [review]
Looks good other than my two remaining suggestions!

**Comment by KrisXV** [comment]
@Zarel what other suggestions? I already addressed them all.

**Comment by Zarel** [comment]
> @Zarel what other suggestions? I already addressed them all.

I'm sorry, I can't tell if you're joking or not. If you're not joking, I was talking about the two you just applied.

**Pull request merged**